torch_geometric/explain/algorithm/base.py

```diff
@@ -37,6 +43,11 @@ def forward(
             target (torch.Tensor): the target of the model.
             target_index (int or torch.Tensor, optional): The target indices to
                 explain. (default: :obj:`None`)
+            node_index (Union[int, Tensor], optional): the node/edge index
+                to explain. Can be a single index if no batch is provided, or a
+                tuple of indices if a batch is provided. only used if the model
+                task level is :obj:`"node"` or :obj:`"edge"`.
+                (default: :obj:`None`)
```

Comment:
What do you mean by "tuple of indices"? I see that a Tensor type is accepted but not a tuple. How exactly does this work for `edge` task level? It would seem to be a bit of a misnomer...

Reply:
This should be replaced by tensor in the docs. For now, the implementation does not work with `edge_level` task, but we could either add an `edge_index` which would be a tuple of node indices in the arguments, or just reuse this (like ` node_index (Union[int, Tensor, Tuple[int,int], List[Tuple[int,int]]]`, where the first two are for selecting nodes, while the second two are for selecting edges. 

And you always know which one you expect with the `model_config.task_level` attributes. What do you think ?

Reply:
I was thinking in the edge explanation setting, we would pass in indices to the edge_index tensor (shape [2, num_edges]). I think it would be confusing to pass in edge index into node_index, so maybe we have separate arguments or rename to a broader term.

Reply:
Oh yes that could make sens. It's probably worth to be in another PR since this is closely related to #5925